src/Kestrel.Core/Internal/AddressBinder.cs

```diff
@@ -19,7 +19,9 @@ internal class AddressBinder
     {
         public static async Task BindAsync(IServerAddressesFeature addresses,
             List<ListenOptions> listenOptions,
+            KestrelServerOptions serverOptions,
```

Comment:
It doesn't look like this is actually used by the DefaultHttpsProvider or anything else. I see that you are setting this on ListenOptions before calling ConfigureHttps, but I don't see why that's necessary. Save some lines of code and don't bother plumbing this through here.

Reply:
It blows up here: https://github.com/aspnet/KestrelHttpServer/blob/83e208436858a2b435e7a9dd0f6ea228bc842e4b/src/Kestrel.Https/ListenOptionsHttpsExtensions.cs#L89

Reply:
I may have misunderstood what you said. There's indeed no need to pass it as an arg to `BindAsync`, since I can get the `KestrelServerOptions` instance from one of the `ListenOptions` in the list. I still need that object though, otherwise a NRE is thrown at the line above.

Reply:
You're right. I forgot about the call to UseHttps().